### DISCUSSION TITLE
OCI-4.6 : Clarification need for 4.7 upgrade document

### DISCUSSION THREAD

**Discussion by ddevadat** [Sunbird-Ed/Community#619] [Documentation] [answered]
We are trying to deploy 4.6 and then upgrade it to 4.7

We are referring from 
https://github.com/project-sunbird/sunbird.org-docs/blob/4.7.0/developer-docs/upgrade/index.md

Need some clear instruction( like full curl command)  on the manual steps given in the documentation for form updates

[SB-28625](https://project-sunbird.atlassian.net/browse/SB-28625)
[SB-28383](https://project-sunbird.atlassian.net/browse/SB-28383)
[SB-28599](https://project-sunbird.atlassian.net/browse/SB-28599)

**Comment by gandham-santhosh** [comment]
These are form configurations. @rajeevsathish  please help.
@ddevadat 4.7 is old version. Any specific reason why you are upgrading from 4.6?

**Reply by ddevadat** [reply]
@gandham-santhosh  there is a requirement to build sunbird on 4.8 , which is why we are doing this approach of 4.6 --> 4.7 --> 4.8

**Reply by ddevadat** [reply]
@rajeevsathish any update on this

**Reply by ddevadat** [reply]
@rajeevsathish could you please update this . This is a show stopper for us

**Comment by rajeevsathish** [comment]
@ddevadat  you need to use the form create api call if the form is not available. or if the form is there then the form update has to be called please refer the following form apis for the same
https://www.postman.com/sunbird-building-blocks/workspace/sunbird-ed/collection/25186239-340f3a7e-ed84-416a-9756-178375ae03b7
hope this helps. but for [SB-28599](https://project-sunbird.atlassian.net/browse/SB-28599) this is not a form but a objectCategory 
please refer this
https://www.postman.com/sunbird-building-blocks/workspace/sunbird-ed/request/25186239-cc920d09-c7f2-4f74-a11e-f7f00ffbc0c5
Regards,
Rajeev
cc: @gandham-santhosh @Krishnaj20 @vinukumar-vs

**Comment by ddevadat** [comment]
Thanks @rajeevsathish . I will try this out and log it here for future reference

**Comment by ddevadat** [comment]
For [SB-28625](https://project-sunbird.atlassian.net/browse/SB-28625)

```
curl --location 'https://{{host}}/api/data/v1/form/update' \
--header 'Content-type: application/json' \
--header 'Authorization: Bearer xyxlkjasldkasdlfknfj' \
--data '{
"request": {
            "type": "user",
            "subType": "framework",
            "action": "update",
            "component": "*",
            "framework": "*",
            "data": {
                "templateName": "defaultTemplate",
                "action": "update",
                "fields": [
                    {
                        "code": "board",
                        "visible": true,
                        "editable": true,
                        "displayProperty": "Editable",
                        "dataType": "text",
                        "renderingHints": {
                            "semanticColumnWidth": "four"
                        },
                        "name": "Board",
                        "description": "Education Board (Like MP Board, NCERT, etc)",
                        "index": 1,
                        "translation": "frmelmnts.lbl.boards",
                        "inputType": "multi-select",
                        "label": "Board",
                        "required": true
                    },
                    {
                        "code": "medium",
                        "visible": true,
                        "editable": true,
                        "displayProperty": "Editable",
                        "dataType": "text",
                        "renderingHints": {
                            "semanticColumnWidth": "four"
                        },
                        "name": "Medium",
                        "description": "Medium of instruction",
                        "index": 2,
                        "translation": "frmelmnts.lbl.medium",
                        "inputType": "multi-select",
                        "label": "Medium",
                        "required": true
                    },
                    {
                        "code": "gradeLevel",
                        "visible": true,
                        "editable": true,
                        "displayProperty": "Editable",
                        "dataType": "text",
                        "renderingHints": {
                            "semanticColumnWidth": "four"
                        },
                        "name": "Class",
                        "description": "Grade",
                        "index": 3,
                        "translation": "frmelmnts.lbl.class",
                        "inputType": "multi-select",
                        "label": "Class",
                        "required": true
                    },
                    {
                        "code": "subject",
                        "visible": true,
                        "editable": true,
                        "displayProperty": "Editable",
                        "dataType": "text",
                        "renderingHints": {
                            "semanticColumnWidth": "four"
                        },
                        "name": "Subject",
                        "description": "Subject of the Content to use to teach",
                        "index": 4,
                        "translation": "frmelmnts.lbl.subject",
                        "inputType": "multi-select",
                        "label": "Subject",
                        "required": false
                    }
                ]
            },
            "created_on": null,
            "last_modified_on": "2021-07-30T12:01:26.239Z",
            "rootOrgId": "*"
        }
}
'
```

**Comment by ddevadat** [comment]
For [SB-26155](https://project-sunbird.atlassian.net/wiki/spaces/MC/pages/3059843073/Form+config+4.6+Browser+Flow+change+story+-+SB-26155)


```
curl --location 'https://{{host}}/api/data/v1/form/create' \
--header 'Content-type: application/json' \
--header 'Authorization: Bearer ljasdlfkalskdf' \
--data '{

    "request": {
         "type": "user",
         "subType":"admin_framework",
         "action":"create",
         "data":{
            "templateName":"defaultTemplate",
            "action":"create",
            "fields":[
               {
                  "code":"board",
                  "dataType":"text",
                  "name":"Board",
                  "label":"Board",
                  "description":"Education Board (Like MP Board, NCERT, etc)",
                  "editable":true,
                  "inputType":"multi-select",
                  "required":true,
                  "displayProperty":"Editable",
                  "visible":true,
                  "renderingHints":{
                     "semanticColumnWidth":"four"
                  },
                  "index":1
               },
               {
                  "code":"medium",
                  "dataType":"text",
                  "name":"Medium",
                  "label":"Medium",
                  "description":"Medium of instruction",
                  "editable":true,
                  "inputType":"multi-select",
                  "required":false,
                  "displayProperty":"Editable",
                  "visible":true,
                  "renderingHints":{
                     "semanticColumnWidth":"four"
                  },
                  "index":2
               },
               {
                  "code":"gradeLevel",
                  "dataType":"text",
                  "name":"Class",
                  "label":"Class",
                  "description":"Grade",
                  "editable":true,
                  "inputType":"multi-select",
                  "required":false,
                  "displayProperty":"Editable",
                  "visible":true,
                  "renderingHints":{
                     "semanticColumnWidth":"four"
                  },
                  "index":3
               },
               {
                  "code":"subject",
                  "dataType":"text",
                  "name":"Subject",
                  "label":"Subject",
                  "description":"Subject of the Content to use to teach",
                  "editable":true,
                  "inputType":"multi-select",
                  "required":false,
                  "displayProperty":"Editable",
                  "visible":false,
                  "renderingHints":{
                     "semanticColumnWidth":"four"
                  },
                  "index":4
               }
            ]
         },
         "created_on":null,
         "last_modified_on":"2021-06-17T06:53:33.442Z",
         "rootOrgId":"*"
      }


}
'
```

**Comment by ddevadat** [comment]
@rajeevsathish 

For 
[SB-28228](https://project-sunbird.atlassian.net/wiki/spaces/MC/pages/3090382849/Form+4.6+-+SB-28228)

I dont see any form to update. Do i have to create the form first? If yes, please provide the payload to create form 

```
    "request": {
        "type": "category",
        "action": "homeListing",
        "subType": "targetedCategory",
        "rootOrgId": "*"
    }
```

**Comment by ddevadat** [comment]
@rajeevsathish 

for [SB-28599](https://project-sunbird.atlassian.net/browse/SB-28599)

in the postman collection, what should be the value for {{obj_cat_name}}
`{{host}}/api/object/category/definition/v1/update/{{obj_cat_name}}_collection_all`

**Reply by vinod303** [reply]
@ddevadat 
in this {{host}}/api/object/category/definition/v1/update/{{obj_cat_name}}_collection_all
here the {{obj_cat_name}}  should the type of "collection" that you want to update.

ie.  if you are trying to update the  object category of collection type Course then
         `/api/object/category/definition/v1/update/obj-cat:course_collection_all`
or
    if you want to update the object of collection type  Content playlist, then it will be
        `/api/object/category/definition/v1/update/obj-cat:content-playlist_collection_all`
or
  if it is for digital textbook
      `/api/object/category/definition/v1/update/obj-cat:digital-textbook_collection_all`
    
And , in the url "all" means , the definition you are creating/updating will affect globally(ie. for all the channels in the system), if you want to this to be set only for a specific channel, then enter the channel id instead of "all"
cc: @rajeevsathish

**Reply by ddevadat** [reply]
i am looking from an upgrade instruction point of view.
[SB-28599](https://project-sunbird.atlassian.net/browse/SB-28599)

in the ticket there is a json attached, i just need to for this json ,what will be object category name

**Reply by vinod303** [reply]
@ddevadat 
`exam-question-set`
Some sample form is available  [here](https://project-sunbird.atlassian.net/browse/SB-28383)
@rajeevsathish

**Comment by ddevadat** [comment]
@rajeevsathish 

For [SB-28383](https://project-sunbird.atlassian.net/browse/SB-28383)

Its not very clear what needs to be done here. My understanding is to create some kind of object category definition. Could you please guide me

**Comment by ashokreddy1208** [comment]
@rajeevsathish Can you please look into this discussion thread

**Comment by ddevadat** [comment]
If there is a delay in getting the ans , we need to know if these steps can be defferred for now and proceed with the actual 4.7 deployment jobs

**Comment by vaivk369** [comment]
@ddevadat For [SB-28383](https://project-sunbird.atlassian.net/browse/SB-28383), the following parameter has to be added/updated in the config->sourcingSettings of the Object Category Definition.
` "downloadCsvEnabled": true`

For the category `exam-question-set` sample form is available [here](https://project-sunbird.atlassian.net/browse/SB-28383) in the attachments.

**Comment by vinukumar-vs** [comment]
@ddevadat I think It was a miss in the release notes(as we are evolving better way to update the release notes).
As commented in the JIRA ticket, you have to update the ObjectCategoryDefination of "exam-question-set" with new property   `"downloadCsvEnabled": true` 
<img width="663" alt="Screenshot 2023-09-26 at 11 11 30 AM" src="https://github.com/Sunbird-Ed/Community/assets/4496096/3e79b3b8-b4b6-4492-a6e6-43b34a31cec3">

Please refer the comments of [SB-28383](https://project-sunbird.atlassian.net/browse/SB-28383) to get more details
https://project-sunbird.atlassian.net/browse/SB-28383?focusedCommentId=93589

@omkar309 @abhay-samagra Can you please confirm the changes.

@vaivk369 @PoojaJManjooran

**Reply by vinukumar-vs** [reply, marked as answer]
@ddevadat As we discussed, form configuration are specially to drive some features on portal/mobile. This is not a blocker for you to  upgrade to 4.7.0.


I hope are you unblocked now as this is not a major blocker for upgrade. If so, can we mark this as answered.
@gandham-santhosh @rajeevsathish @Krishnaj20

**Reply by ddevadat** [reply]
will park it now as its understood that its not a show stopper
